basestar-storage/src/test/java/io/basestar/storage/TestStorage.java

```diff
@@ -761,15 +759,54 @@ protected void testLarge() {
         data.put("string", str.toString());
         final Instance instance = instance(schema, id, 1L, data);
 
-        storage.write(Consistency.ATOMIC, Versioning.CHECKED)
-                .createObject(schema, id, instance)
-                .write().join();
+        try {
+            storage.write(Consistency.ATOMIC, Versioning.CHECKED)
+                    .createObject(schema, id, instance)
+                    .write().join();
+
+            final BatchResponse results = storage.read(Consistency.ATOMIC)
+                    .get(schema, id, ImmutableSet.of())
+                    .read().join();
+
+            assertEquals(1, results.getRefs().size());
+        } catch (final ObjectSizeException e) {
+            log.warn("Suppressed", e);
+        }
+    }
+
+    @Test
+    protected void testLargeIndex() {
+
+        assumeTrue(supportsOversize());
+
+        final Storage storage = storage(namespace);
+
+        final ObjectSchema schema = namespace.requireObjectSchema(ADDRESS);
+
+        final String id = UUID.randomUUID().toString();
+
+        final StringBuilder str = new StringBuilder();
+        for (int i = 0; i != 1000000; ++i) {
+            str.append("test");
+        }
 
-        final BatchResponse results = storage.read(Consistency.ATOMIC)
-                .get(schema, id, ImmutableSet.of())
-                .read().join();
+        final Map<String, Object> data = new HashMap<>();
+        data.put("country", str.toString());
+        final Instance instance = instance(schema, id, 1L, data);
 
-        assertEquals(1, results.getRefs().size());
+        try {
+            storage.write(Consistency.ATOMIC, Versioning.CHECKED)
+                    .createObject(schema, id, instance)
+                    .write().join();
+
+            final BatchResponse results = storage.read(Consistency.ATOMIC)
+                    .get(schema, id, ImmutableSet.of())
+                    .read().join();
+
+            assertEquals(1, results.getRefs().size());
+        } catch (final IndexRecordSizeException e) {
+            log.warn("Suppressed", e);
```

Comment:
no, because either outcome is acceptable

storage is allowed to persist a big object record or a big index record, but if it can't then it must throw this exception rather than some internal storage-driven exception